### PR TITLE
RUN-4942 Added proxy support for Mac OS

### PR DESCRIPTION
**Pull request by luiemilio**

### CODE REVIEW

File: src/launcher/util.ts [hunk 1]
```diff
@@ -14,9 +14,49 @@ export async function exists(path: string): Promise<Boolean> {
...
+        port: parsedUrl.port,
+        host: parsedUrl.hostname,
+        username: parsedUrl.username,
+        password: parsedUrl.password
+    };
+}
+
+function proxyExists() {
+    return process.env.HTTPS_PROXY || process.env.https_proxy;
+}
```
Comment (licui3936) [inline src/launcher/util.ts]: How about process.env.HTTP_PROXY/http_proxy?

Comment (luiemilio) [reply src/launcher/util.ts]: We are making an https request to get resolve the runtime version and download it, so I figured we would need to only care about HTTPS_PROXY being set.

Comment (licui3936) [reply src/launcher/util.ts]: gotcha

**Comment by datamadic** [comment]
Revisit in .41 all you Luis!

**Pull request closed without merging**